### PR TITLE
Replace ButtonBox with Box

### PR DESCRIPTION
**Pull request by ZenitsuDev**
This aims to replace ButtonBox with Box for GTK4 porting. The dimensions of the action buttons are made the same to those buttons inside a buttonbox for similarity

### CODE REVIEW

File: src/Bubble.vala [hunk 1]
```diff
@@ -174,15 +174,16 @@ public class Notifications.Bubble : AbstractBubble {
             attach (body_label, 1, 1);
 
             if (notification.actions.length > 0) {
-                var action_area = new Gtk.ButtonBox (Gtk.Orientation.HORIZONTAL) {
-                    layout_style = Gtk.ButtonBoxStyle.END
-                };
+                var action_area = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 0);
```
Comment (danrabbit) [inline src/Bubble.vala:177]: One property that buttonbox has by default that we don't get here for free is `homogeneous`:
```suggestion
                var action_area = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 0) {
                    homogeneous = true
                };
```

File: src/Bubble.vala [hunk 2]
```diff
@@ -174,15 +174,16 @@ public class Notifications.Bubble : AbstractBubble {
...
 
                 bool action_area_packed = false;
 
                 for (int i = 0; i < notification.actions.length; i += 2) {
                     if (notification.actions[i] != "default") {
-                        var button = new Gtk.Button.with_label (notification.actions[i + 1]);
+                        var button = new Gtk.Button.with_label (notification.actions[i + 1]) {
+                            width_request = 85,
+                            height_request = 27
```
Comment (danrabbit) [inline src/Bubble.vala:185]: We probably shouldn't hardcode these values because the UI can scale when text size changes, so this would break that

**Review by danrabbit — CHANGES_REQUESTED** [review]
Left a couple of inline comments!

Instead of hardcoding the button size values, we should set `homogeneous` on the button box.

We also need to update the CSS since we use that to add margins and spacing. So probably the best way to do that is to add a css class "buttonbox" and then update Application.css anywhere it says `buttonbox` to `.buttonbox`

File: src/Bubble.vala (hunk 1 above)
Comment (ZenitsuDev) [reply src/Bubble.vala:177]: oh yeah right.. lemme commit this. Thanks!

File: src/Bubble.vala (hunk 2 above)
Comment (ZenitsuDev) [reply src/Bubble.vala:187]: Hi! Should we set it then as minimum through css? I tried without the width and height request property and it looks quite different as it is smaller than buttons inside buttonbox.

Comment (danrabbit) [reply src/Bubble.vala]: I pushed a commit setting min-width in CSS. I think probably what should happen is this style class should move to the system stylesheet, so I'll open a branch for that

**Review by danrabbit — APPROVED** [review]
Let's go ahead and get this merged :+1:

**Pull request merged**